EmeraldComponents/Basics/Label/EmeraldLabel.swift

```diff
@@ -0,0 +1,305 @@
+            switch state {
+            case .error:
+                return UIColor.Palette.Basic.error
+            case .warning:
+                return UIColor.Palette.Basic.warning
+            case .neutral:
+                return UIColor.Palette.Light.white4
+            case .success:
+                return UIColor.Palette.Basic.success
+            case .info:
+                return UIColor.Palette.purple
+            }
+        default:
+            return UIColor.clear
+        }
+    }
+    
+    /// Sets label border with to be displayed
+    ///
+    /// - Parameters:
+    ///   - state: Label state enum
+    ///   - type: Label type enum
+    /// - Returns: Return label border width
+    internal func borderWidth(forState state: EmeraldLabelState, andType type: EmeraldLabelType) -> CGFloat {
+        switch type {
+        case .outline:
+            return 1.0
+        default:
+            return 0.0
+        }
+    }
+    
+    /// Sets border label border color to be displayed
+    ///
+    /// - Parameters:
+    ///   - state: Label state enum
+    ///   - type: Label type enum
+    /// - Returns: Return label border color
+    internal func borderColor(forState state: EmeraldLabelState, andType type: EmeraldLabelType) -> CGColor {
+        switch type {
+        case .fill, .outline:
+            switch state {
+            case .error:
+                return UIColor.Palette.Basic.error.cgColor
+            case .warning:
+                return UIColor.Palette.Basic.warning.cgColor
+            case .neutral:
+                return UIColor.Palette.Light.white4.cgColor
+            case .success:
+                return UIColor.Palette.Basic.success.cgColor
+            case .info:
+                return UIColor.Palette.purple.cgColor
+            }
+        default:
+            return UIColor.clear.cgColor
+        }
+    }
+    
+    /// Sets an image according to label type. I'll always displays a bullet fot text labe.
+    ///
+    /// - Parameters:
+    ///   - state: Label state enum
+    ///   - type: Label type enum
+    /// - Returns: Return an label image to be displayed
+    internal func textIcon(forState state: EmeraldLabelState, andType type: EmeraldLabelType) -> UIImage? {
+        switch type {
+        case .text:
+            return UIImage(named: "icon-dot", in: Bundle.basic, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        case .image:
+            return UIImage(named: auxIconLabel, in: Bundle.basic, compatibleWith: nil)!.withRenderingMode(.alwaysTemplate)
+        default:
+            return nil
+        }
+    }
+    
+    /// Set constraints visibility according to label type, if label type there were no icons to display, so constraints needs to be 0.0
+    ///
+    /// - Parameters:
+    ///   - state: Label state enum
+    ///   - type: Label type enum
+    internal func constraintsVisibility(forState state: EmeraldLabelState, andType type: EmeraldLabelType) {
```

Comment:
```suggestion
    internal func setVisibilityConstraints(forState state: EmeraldLabelState, andType type: EmeraldLabelType) {
```
I think this naming for the method would represent what it does more clearly.